app/models/task.rb

```diff
@@ -77,9 +77,9 @@ class Task < ActiveRecord::Base
 
   # Ensures that the control pin and the power pin are set to different values or no value.
   def event_and_parent_validation
-    if parent_id.nil? ^ trigger.nil?
-      errors.add(:trigger, ' may be not set unless the Parent Task is specified.') if parent_id.nil?
-      errors.add(:parent_id, ' requires a trigger event.') if trigger.nil?
+    unless schedule.nil? or schedule.root_task_id.nil?
```

Comment:
Keep the old validation and make a second validation that handles this case.

Additionally, don't use the `or` method. Just use `||`. Less of an issue here, but `or` has lower operator precedence and is very rarely useful.